### PR TITLE
Hide legacy proposals #1986

### LINKED ISSUE #23: [4] Handle proposals related to phishing accounts
Original issue #1755

- [x] Add toggle in Header of "Proposed transactions": "Deactivate suspicious proposals"
- [x] List of phishing accounts is in app/lib/common/scamAccounts.js, plus blacklist of user (see Burger > Settings > Whitelist)
- [x] If toggle is on, deactive the "Approve" button of such proposals (option to permanently reject remains!)

### PR DESCRIPTION
**Pull request by VerevkinAlexander**
Closes #23

.I added new method hideProposals() for checking and hiding legacy proposals,
also I added checkbox and hideLegacyProposals state

### CODE REVIEW

**Comment by VerevkinAlexander** [comment]
@sschiessl-bcp Please review it , when you have time.

**Comment by sschiessl-bcp** [comment]
Please reference the corresponding issue in the PR

**Comment by VerevkinAlexander** [comment]
Closes https://github.com/blockchainprojects/bitshares-ui/issues/20

**Comment by sschiessl-bcp** [comment]
The correct issue is the one within this repository, as this is the issue that will be closed through this PR. I have edited your comment above.

**Comment by sschiessl-bcp** [comment]
I think the recognition is off.

Please adjust:
 - [ ] Proposals with almost same name and same payout account.
   - [ ] if both are not active, hide older one
   - [ ] if one is active, hide other one.  Example: 1.14.74 and 1.14.73. 
- [ ] Add a manually defined list of ids that are to be hidden: For now: ["1.14.114","1.14.107"]
- [ ] If the start date of the proposal is more than one month in the past, and less than 10 million votes

**Comment by VerevkinAlexander** [comment]
> if one is active, hide other one. Example: 1.14.74 and 1.14.73.

I am not sure that understood correctly. Should we display some active proposals when hide proposals is enabled?

**Comment by sschiessl-bcp** [comment]
>  if one is active, hide other one. Example: 1.14.74 and 1.14.73.

Consider the header of the proposal tab
image

Listed under "Proposed": Inactive
Listed under "Active": Active

**Comment by VerevkinAlexander** [comment]
I adjusted filter according to your`s comments. Please review it when you have time.

File: app/components/Account/AccountVoting.jsx [hunk 1]
```diff
@@ -549,13 +552,73 @@ class AccountVoting extends React.Component {
         let workerArray = this._getWorkerArray();
 
         let voteThreshold = 0;
+        const hideProposals = filteredWorker => {
+            const isActive = worker => {
+                return (
```
Comment (sschiessl-bcp) [inline app/components/Account/AccountVoting.jsx]: An account is considerd active when its within the workers list of line 615

File: app/components/Account/AccountVoting.jsx [hunk 2]
```diff
@@ -757,6 +823,21 @@ class AccountVoting extends React.Component {
             </div>
         );
 
+        const hideLegacy = (
+            <div>
+                <input
```
Comment (sschiessl-bcp) [inline app/components/Account/AccountVoting.jsx]: Please use a Switch from style-guide
`import {Switch} from "bitshares-ui-style-guide";`

For usage see e.g. https://github.com/blockchainprojects/bitshares-ui/blob/8f8cf007d4ed8aa7cb83aec018bf0b964cee502a/app/components/Exchange/Personalize.jsx#L220

If possible, put this switch in line on the right side with this 
image

Furthermore, please give the switch a onHover popup, I will include text then what "legacy" means

**Review by sschiessl-bcp — CHANGES_REQUESTED** [review]

File: app/components/Account/AccountVoting.jsx (hunk 1 above)
Comment (VerevkinAlexander) [reply app/components/Account/AccountVoting.jsx]: fixed

File: app/components/Account/AccountVoting.jsx (hunk 2 above)
Comment (VerevkinAlexander) [reply app/components/Account/AccountVoting.jsx]: fixed

**Comment by sschiessl-bcp** [comment]
Accepted

**Pull request closed without merging**